packages/core/src/odata/v2/request-builder/update-request-builder.ts

```diff
@@ -91,20 +102,45 @@ export class UpdateRequestBuilderV2<EntityT extends EntityV2>
 
     return (
       this.build(destination, options)
+        .then(request => this.warnIfNavigation(request))
         .then(request => request.execute())
 
         // Update returns 204 hence the data from the request is used to build entity for return
-        .then(response =>
-          this._entity
+        .then(response => {
+          const eTag =
+            extractEtagFromHeader(response.headers) ||
+            extractODataEtagV2(response.data) ||
+            this.requestConfig.eTag;
+          return this._entity
             .setOrInitializeRemoteState()
-            .setVersionIdentifier(this.requestConfig.eTag)
-        )
+            .setVersionIdentifier(eTag);
+        })
         .catch(error =>
           Promise.reject(errorWithCause('OData update request failed!', error))
         )
     );
   }
 
+  /*
+   * In case the entity contains a navigation to a different entity a warning is printed.
+   */
+  warnIfNavigation(
+    request: ODataRequest<ODataUpdateRequestConfig<EntityT>>
+  ): ODataRequest<ODataUpdateRequestConfig<EntityT>> {
+    Object.keys(this._entity).forEach(key => {
+      if (
+        this._entity[key] instanceof EntityV2 ||
+        (this._entity[key] && this._entity[key][0] instanceof EntityV2)
+      ) {
+        logger.warn(
```

Comment:
Maybe we should rather collect all navigation properties and only log them once all together?

Reply:
yes